client/blocks/signup-form/passwordless.jsx

```diff
@@ -0,0 +1,189 @@
+/**
+ * External dependencies
+ */
+import React, { Component } from 'react';
+import { connect } from 'react-redux';
+import { localize } from 'i18n-calypso';
+import emailValidator from 'email-validator';
+
+/**
+ * Internal dependencies
+ */
+import Button from 'components/button';
+import FormLabel from 'components/forms/form-label';
+import FormTextInput from 'components/forms/form-text-input';
+import LoggedOutForm from 'components/logged-out-form';
+import LoggedOutFormFooter from 'components/logged-out-form/footer';
+import ValidationFieldset from 'signup/validation-fieldset';
+import { onboardPasswordlessUser } from 'lib/signup/step-actions';
+import { recordTracksEvent } from 'state/analytics/actions';
+import Notice from 'components/notice';
+import { submitSignupStep } from 'state/signup/progress/actions';
+
+export class PasswordlessSignupForm extends Component {
+	static defaultProps = {
+		locale: 'en',
+	};
+
+	state = {
+		isSubmitting: false,
+		email: '',
+		errorMessages: null,
+	};
+
+	submitTracksEvent = ( isSuccessful, message ) =>
+		this.props.recordTracksEvent( 'calypso_signup_actions_onboarding_passwordless_login', {
+			is_successful: isSuccessful,
```

Comment:
I'm not sure, but it may make it easier for us to create funnels, etc. if we use separate event names for successful & unsuccessful attempts (rather than putting that information in event props).